### PR TITLE
Update documentation for manual masks creation

### LINKED ISSUE #336: Add documentation section on how to create mask
To clarify how masks should be done (e.g. avoid this situation https://github.com/neuropoly/axondeepseg/issues/329#issuecomment-661021032), we should add a comprehensive section in RTD; things doc could include:
- put it there: Help/Create mask
- include image: bad mask / good mask (use the same mask, and fix the issues)
- mention about the shades of gray (even after fixing #335)

### PR DESCRIPTION
**Pull request by mariehbourget**
Fixes #336

### CODE REVIEW

File: docs/source/documentation.rst [hunk 1]
```diff
@@ -404,21 +404,69 @@ Help
 
 If you experience issues during installation and/or use of AxonDeepSeg, you can post a new issue on the `AxonDeepSeg GitHub issues webpage <https://github.com/neuropoly/axondeepseg/issues>`_. We will reply to you as soon as possible.
 
-Manual correction
+Guide for manual labelling
+===============================================================================
```
Comment (jcohenadad) [inline docs/source/documentation.rst:408]: sphinx requirement: underline should be same size as subheading number of characters

File: docs/source/documentation.rst [hunk 2]
```diff
@@ -404,21 +404,69 @@ Help
...
+    :width: 750px
+    :align: center
+    :height: 500px
+    :alt: Bad mask example
+    
+    Bad mask example
+
+To create a manual mask for training, you can try one of the following:
+
+* Try segmenting your image with AxonDeepSeg default SEM or TEM models and make manual corrections of the segmentation mask in FSLeyes or GIMP software.
```
Comment (jcohenadad) [inline docs/source/documentation.rst:447]:
```suggestion
* Try segmenting your images with AxonDeepSeg's default models and make manual corrections of the segmentation masks in FSLeyes or GIMP software (see section below).
```

File: docs/source/documentation.rst [hunk 3]
```diff
@@ -404,21 +404,69 @@ Help
...
+
+Manual correction of segmentation masks
 -------------------------------------------------------------------------------
 
 If the segmentation with AxonDeepSeg fails or does not give optimal results, you can try one of the following options:
 
-**Option 1: manual correction of the segmentation masks**
+**Option 1: manual correction of the segmentation masks with FSLeyes**
+
+* In FSLeyes, load your image and apply ADS prediction with the default SEM or TEM models.
```
Comment (jcohenadad) [inline docs/source/documentation.rst:458]: "apply ADS prediction" --> no, this was already done in the previous section. Leave it here and change previous section, or change text here

File: docs/source/documentation.rst [hunk 4]
```diff
@@ -404,21 +404,69 @@ Help
...
+* Make correction on myelin and axons masks using the Edit Tool in **Tools > Edit mode**.
```
Comment (jcohenadad) [inline docs/source/documentation.rst:459]: i think the idea is to *only* correct myelin mask, not axons (use filling tool for axons)-- if we ask users to correct both, they will make mistakes (e.g. missing pixels at the interface)

File: docs/source/documentation.rst [hunk 5]
```diff
@@ -404,21 +404,69 @@ Help
...
 
-* After correcting the myelin mask, you can regenerate the segmentation masks (axon+myelin). To do this, you can use the following notebook: https://github.com/neuropoly/axondeepseg/blob/master/notebooks/generate_axons_from_myelin.ipynb.
+* Note that when you launch a segmentation, in the folder output, you will also find the axon and myelin masks (with the suffixes **'_seg-axon.png'** and **'_seg-myelin.png'**). If the segmentation proposed by AxonDeepSeg is not optimal, you can manually correct the myelin mask and create a new axon+myelin mask.
+* For the manual correction or the creation of a new manual mask, we suggest using the GIMP software (https://www.gimp.org/). For a more detailed procedure, please consult the following link: `Manual labelling with GIMP <https://docs.google.com/document/d/10E6gzMP6BNGJ_7Y5PkDFmum34U-IcbMi8AvRruhIzvM/edit>`_.
 
-**Option 2: manual correction combined with *AxonSeg* software**
+**Option 3: manual correction combined with *AxonSeg* software**
 
-* Manually correct the axon mask (as explained in Option 1).
+* Manually correct the axon mask (as explained in Option 2).
```
Comment (jcohenadad) [inline docs/source/documentation.rst:469]: i would remove

File: docs/source/documentation.rst (hunk 1 above)
Comment (mariehbourget) [reply docs/source/documentation.rst:408]: I found that the requirement is for the underline to be "at least as long as the text" ([ref](https://www.sphinx-doc.org/en/1.8/usage/restructuredtext/basics.html#sections)). I don't mind changing it to the same length but I have to change all the other underlines in the document.

File: docs/source/documentation.rst (hunk 5 above)
Comment (mariehbourget) [reply docs/source/documentation.rst]: Resolved in b8a2963

File: docs/source/documentation.rst (hunk 3 above)
Comment (mariehbourget) [reply docs/source/documentation.rst]: Resolved in e241f51

File: docs/source/documentation.rst (hunk 4 above)
Comment (mariehbourget) [reply docs/source/documentation.rst]: Resolved in e241f51

File: docs/source/documentation.rst (hunk 1 above)
Comment (mariehbourget) [reply docs/source/documentation.rst]: Resolved in 5d8d474

**Comment by mariehbourget** [comment]
This PR is ready for review.

In addition to the changed files on github, I updated the GIMP procedure (linked in our doc) to create or correct a myelin mask and fill the axons [here](https://docs.google.com/document/d/10E6gzMP6BNGJ_7Y5PkDFmum34U-IcbMi8AvRruhIzvM/edit#heading=h.qb69k75v999j). 
@Stoyan-I-A or @mathieuboudreau, could you review the procedure and let me know if it is ok (you should have access to comment on the doc). Thanks!

**Comment by Stoyan-I-A** [comment]
The documentation looks good.
I will try the GIMP procedure soon.

**Comment by jcohenadad** [comment]
agreed about the video tutorials. people LOVE them ❤️

**Comment by mariehbourget** [comment]
Thanks @mathieuboudreau for the great feedback, it helps a lot to have a new perspective.
I'll work on this PR as a priority.
Option 1: I agree with you, I think the best way to go is to make a complete detailed procedure as we have with GIMP.
Option 2: The link to the Gimp website was intended as a download link (not for general documentation). I'll clarify that. Also, I like the idea to have separate Google docs for procedures that are not "ADS". I think they are easier to maintain overtime to with several screenshots involved.

**Comment by mariehbourget** [comment]
Hi @mathieuboudreau, I've updated the RTD parts ans the GIMP procedure according to your comments.
I still have to write the detailed procedure for FSLeyes.

**Comment by mariehbourget** [comment]
The detailed procedure for FSLeyes is also completed. This PR is ready for re-review.

**Comment by Stoyan-I-A** [comment]
I tested the GIMP procedure, it worked well 👍.

**Comment by mariehbourget** [comment]
Thanks @Stoyan-I-A . There is also a complete procedure for FSLeyes if you want to take a look ([here](https://docs.google.com/document/d/1S8i96cJyWZogsMw4RrlQYwglcOWd3HrM5bpTOJE4RBQ/edit#heading=h.qb69k75v999j)).

**Comment by mathieuboudreau** [comment]
Thanks @mariehbourget!

I will try and re-review tonight, or at the latest sometime tomorrow.

**Comment by mariehbourget** [comment]
Thanks @mathieuboudreau for the review.
And yes, your mask is good :)

**Comment by mathieuboudreau** [comment]
Thanks! 

That reminds me, we should offer a "validator" that will check if your masks is the correct format, etc. I'll open an issue about it.

**Comment by mariehbourget** [comment]
Yes, that could be included in #335.

**Pull request merged**